courses/models.py

```diff
@@ -309,35 +318,58 @@ def __str__(self):
         return self.title
 
 
-class CourseRunEnrollment(TimestampedModel, AuditableModel):
-    """
-    Link between User and CourseRun indicating a user's enrollment
-    """
+class EnrollmentModel(TimestampedModel, AuditableModel):
+    """Abstract base model for enrollments"""
+
+    class Meta:
+        abstract = True
 
     user = models.ForeignKey(User, on_delete=models.PROTECT)
-    run = models.ForeignKey("courses.CourseRun", on_delete=models.PROTECT)
     company = models.ForeignKey(
         "ecommerce.Company", null=True, on_delete=models.PROTECT
     )
-    edx_enrolled = models.BooleanField(
-        default=False,
-        help_text="Indicates whether or not the request succeeded to enroll via the edX API",
+    change_status = models.CharField(
+        choices=ENROLL_CHANGE_STATUS_CHOICES, max_length=20, null=True, blank=True
     )
     active = models.BooleanField(
         default=True,
         help_text="Indicates whether or not this enrollment should be considered active",
     )
 
-    class Meta:
-        unique_together = ("user", "run")
+    active_objects = ActiveEnrollmentManager()
```

Comment:
As discussed, this should replace the default `objects` manager and we'll want one for `all_objects` too, which probably requires an override of the django admin `get_queryset` since we want all the records in there I think?

Since this is defined on an abstract parent model class, can you update the docstrings on each of the subclassing models calling this out with an example as the way we want to always query for these?